backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -357,13 +366,78 @@ object Compiler {
 
     val uniqueInputs = compileInputs.uniqueInputs
     reporter.reportStartCompilation(previousProblems)
+    val fileManager = newFileManager
+
+    // Manually skip redundant best-effort compilations. This is necessary because compiler
+    // phases supplying the data needed to skip compilations in zinc remain unimplemented for now.
+    val noopBestEffortResult = compileInputs.previousCompilerResult match {
+      case Failed(
+            problems,
+            t,
+            elapsed,
+            _,
+            bestEffortProducts @ Some(BestEffortProducts(compileProducts, previousHash))
```

Comment:
```suggestion
            bestEffortProducts @ Some(BestEffortProducts(previousCompilationResults, previousHash))
```
minor, but this seems misleading down since you delete newClassesDir and then copy from compileProducts.newClassesDir

And those are two different directories, right?  You delete the current directory and copy over old classes dir?

Reply:
Yes, `newClassesDir` Is a newly created directory for new scalac compilation artifacts (which we will not have in this case so we delete it). The we copy to the previous (complete) results to clientDir (useful when we e.g restart metals and it's assigned a new empty directory)